pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector.go

```diff
@@ -109,21 +164,58 @@ func parseLogs(logs []string, metrics []string, filters []string) (*v1beta1.Obse
 	}
 	// If objective metrics were not reported, insert unavailable value in the DB
 	if !isObjectiveMetricReported {
-		olog.MetricLogs = []*v1beta1.MetricLog{
-			{
-				TimeStamp: time.Time{}.UTC().Format(time.RFC3339),
-				Metric: &v1beta1.Metric{
-					Name:  metrics[0],
-					Value: consts.UnavailableMetricValue,
+		klog.Infof("Objective metric %v is not found in training logs, %v value is reported", metrics[0], consts.UnavailableMetricValue)
+		return &v1beta1.ObservationLog{
+			MetricLogs: []*v1beta1.MetricLog{
+				{
+					TimeStamp: time.Time{}.UTC().Format(time.RFC3339),
+					Metric: &v1beta1.Metric{
+						Name:  metrics[0],
+						Value: consts.UnavailableMetricValue,
+					},
 				},
 			},
 		}
-		klog.Infof("Objective metric %v is not found in training logs, %v value is reported", metrics[0], consts.UnavailableMetricValue)
-	} else {
-		olog.MetricLogs = mlogs
 	}
+	return &v1beta1.ObservationLog{
+		MetricLogs: mlogs,
+	}
+}
+
+func parseTimestamp(timestamp interface{}) string {
+	if stringTimestamp, ok := timestamp.(string); ok {
+
+		if _, err := time.Parse(time.RFC3339Nano, stringTimestamp); err != nil {
+			return ""
+		} else {
+			return stringTimestamp
+		}
 
-	return olog, nil
+	} else {
```

Comment:
Do we want to support other timestamp, not in `RFC3339 ` format ? 
I believe for TEXT we support only 1 format of timestamp.
/cc @johnugeorge @gaocegege

Reply:
Since `cloudml-hypertune` uses UNIX time as a timestamp, I implemented it so that `file-metrics-collector` can parse UNIX time.

https://github.com/GoogleCloudPlatform/cloudml-hypertune/blob/7374cae8b0285c1ee7697f7aaed6f42b267e2257/hypertune/hypertune.py#L69

Reply:
I see. Do you know how our Katib UI parses this Trial time data ?
cc @kimwnasptd @seong7

Reply:
It seems that Katib UI parses `ObservationLog.MetricLogs[*].TimeStamp` in `RFC3339Nano` format.

https://github.com/kubeflow/katib/blob/fa1babe3c8ad1d9b16ea6bbdba12ea1eb9857e64/pkg/new-ui/v1beta1/hp.go#L186-L188

My `parseTimestamp` function converts the timestamp in UNIX time format to `RFC3339Nano` format and stores it in `ObservationLog.MetricLogs[*].TimeStamp`, so I don't think it will affect our Katib UI.

https://github.com/kubeflow/katib/blob/c69c6e1db20ba86ec54872cc09fe26f3a312c846/pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector.go#L215

WDYT? @andreyvelich

Reply:
@tenzen-y correct, the UI handles RFC3339 timestamps. The parsing logic looks good, so I also don't think we'll have an issue with the UI

Reply:
Thank you for letting me know! @kimwnasptd

Reply:
Sounds good @tenzen-y.
Should we add Warning messages before return empty string (e.g. [here](https://github.com/kubeflow/katib/blob/87d2a39d2bb46b87b0e4cc3660a93ec673adb1ee/pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector.go#L189))

Reply:
Done.